### PR TITLE
Chore: Upgrades Cypress to 7.5.0

### LINKED ISSUE #35420: Upgrade Cypress
We are still running on a 6.x version of Cypress so we should upgrade to latest as there are a lot of fixes!

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
For details on the migration see: https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-7-0

**Which issue(s) this PR fixes**:
Fixes #35420 

**Special notes for your reviewer**:

### CODE REVIEW

File: e2e/suite1/specs/dashboard-time-zone.spec.ts [hunk 1]
```diff
@@ -1,82 +1,82 @@
...
-
-const isTimeCorrect = (utc: string, tz: string, offset: number): boolean => {
-  const minutes = 1000 * 60;
-
-  const a = Cypress.moment(utc, 'HH:mm').set('seconds', 0).set('milliseconds', 0);
-
-  const b = Cypress.moment(tz, 'HH:mm').set('seconds', 0).set('milliseconds', 0).add('hours', offset);
-
-  return a.diff(b, 'minutes') <= 6 * minutes;
-};
```
Comment (hugohaggmark) [inline e2e/suite1/specs/dashboard-time-zone.spec.ts:82]: @mckn I had to skip comment out this test for 2 reasons:

1. It doesn't catch the current bug https://github.com/grafana/grafana/issues/35504
2. Cypress.moment has been removed so this resulting in a lot of transpile errors

Would be great if you can have a look. Would be great if we can catch the bug and not use moment.

File: packages/grafana-e2e/src/flows/configurePanel.ts [hunk 2]
```diff
@@ -114,11 +114,9 @@ export const configurePanel = (config: PartialAddPanelConfig | PartialEditPanelC
       }
     }
 
-    e2e().server();
-
     // @todo alias '/**/*.js*' as '@pluginModule' when possible: https://github.com/cypress-io/cypress/issues/1296
 
-    e2e().route(chartData.method, chartData.route).as('chartData');
+    e2e().intercept(chartData.method, chartData.route).as('chartData');
```
Comment (hugohaggmark) [inline packages/grafana-e2e/src/flows/configurePanel.ts:119]: Those this cause a lot of trouble for you @vickyyyyyyy? I can't test this.

**Comment by hugohaggmark** [comment]
@zoltanbedi there are a couple of e2e tests that uses intercept that no longer work with this upgrade, can you have a look and push changes to this PR. Thank you!

File: packages/grafana-e2e/src/flows/configurePanel.ts (hunk 2 above)
Comment (vickyyyyyyy) [reply packages/grafana-e2e/src/flows/configurePanel.ts:119]: I don't think so - I ran our enterprise plugins e2e tests with this and it looks fine 👍

**Review by vickyyyyyyy — APPROVED** [review]
LGTM! Ran this with our enterprise plugins and all looks good 🔥

File: e2e/suite1/specs/exemplars.spec.ts [hunk 3]
```diff
@@ -35,16 +35,16 @@ describe('Exemplars', () => {
...
-    e2e().intercept('POST', '/api/v1/query_range', {
+    e2e().intercept('POST', '**/api/v1/query_range', {
       fixture: 'prometheus-query-range-response.json',
     });
-    e2e().intercept('POST', '/api/v1/query', {
+    e2e().intercept('POST', '**/api/v1/query', {
       fixture: 'prometheus-query-response.json',
     });
-    e2e().intercept('POST', '/api/ds/query', {
+    e2e().intercept('POST', '**/api/ds/query', {
```
Comment (hugohaggmark) [inline e2e/suite1/specs/exemplars.spec.ts:47]: @zoltanbedi did you read https://docs.cypress.io/guides/references/migration-guide#cy-intercept-changes? Not sure this affect these, just making sure as the order here seems to be the same :)

Comment (zoltanbedi) [reply e2e/suite1/specs/exemplars.spec.ts:47]: yep I saw it, I think it won't effect it.

**Review by mckn — APPROVED** [review]
Great work with this! LGTM

File: e2e/suite1/specs/exemplars.spec.ts (hunk 3 above)
Comment (hugohaggmark) [reply e2e/suite1/specs/exemplars.spec.ts:47]: 👍

**Review by zoltanbedi — APPROVED** [review]
Awesome

**Comment by hugohaggmark** [comment]
@mckn e2e is failing for the timezone test, great work on that! (which is correct as we have a bug currently) Waiting for #35680 to make this PR green again :)

**Pull request merged**